pkg-config files on >=10.14 assume headers in /usr/include

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?
```
% brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The following directories do not exist:
/usr/local/sbin

You should create these directories and change their ownership to your account.
  sudo mkdir -p /usr/local/sbin
  sudo chown -R $(whoami) /usr/local/sbin
% brew config
HOMEBREW_VERSION: 1.7.6-143-gc6acab6
ORIGIN: https://github.com/Homebrew/brew
HEAD: c6acab66feb63743fe69b38dab7dc7b90710e838
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 537f3703aa41c550edbaaf28dd280c6ec929f946
Core tap last commit: 34 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14-x86_64
CLT: 10.0.0.0.1.1535735448
Xcode: 10.0
XQuartz: 2.7.11 => /opt/X11
```

To help us debug your issue please explain:
- What you were trying to do (and why)
I tried to build a package which uses `pkg-config` from Homebrew.

- What happened (include command output)
I met an error when build:
```
fatal error: 'libxml/tree.h' file not found
#include <libxml/tree.h>
         ^~~~~~~~~~~~~~~
1 error generated.
```

It looks like `pkg-config` from Homebrew is not providing the correct include directory of libs shipping with macOS(Xcode and CLI tools). As the release note of Xcode 10, header files which were under `/usr/include` no longer exists and been moved to Xcode: `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include` and CLI tools: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include`.

output:
```
% which pkg-config
/usr/local/bin/pkg-config

% pkg-config --cflags --libs "libxml-2.0 > 2.6.17"
-I/usr/include/libxml2 -lxml2

% pkg-config --variable pcfiledir "libxml-2.0 > 2.6.17"
/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14

% cat /usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14/libxml-2.0.pc 
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include
modules=1

Name: libXML
Version: 2.9.4
Description: libXML library version2.
Requires:
Libs: -L${libdir} -lxml2
Libs.private: -lz -lpthread -licucore -lm
Cflags: -I${includedir}/libxml2
```

I tested only with `libxml2` but I believe this issue affect all libs which have `pkg-config` file from Homebrew `Library/Homebrew/os/mac/pkgconfig/10.14`, they are: libcurl.pc, libexslt.pc, libxml-2.0.pc, libxslt.pc, sqlite3.pc, zlib.pc.
- What you expected to happen
```
% pkg-config --cflags --libs "libxml-2.0 > 2.6.17"
```

should gives the correct path to the header files under Xcode or CLI tools. 

- Step-by-step reproduction instructions (by running `brew` commands)
1. `brew install pkg-config`
2. `pkg-config --cflags --libs "libxml-2.0 > 2.6.17"`

Also, setting `PKG_CONFIG_SYSROOT_DIR` globally can cause other pkg-config files to fail, as I just noticed as I moved on to another task in the same terminal that I did this testing:

~~~
CMake Error in src/CMakeLists.txt:
  Imported target "ignition-common3::requested" includes non-existent path

    "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/local/Cellar/ossp-uuid/1.6.2_2/include/ossp"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
~~~

--- a/Library/Homebrew/extend/os/mac/extend/ENV/super.rb
+++ b/Library/Homebrew/extend/os/mac/extend/ENV/super.rb
@@ -112,6 +112,8 @@ module Superenv
     end
     generic_setup_build_environment(formula)
 
+    self["PKG_CONFIG"] = Superenv.bin/"pkg-config"
+
     # Filter out symbols known not to be defined since GNU Autotools can't
     # reliably figure this out with Xcode 8 and above.
     if MacOS.version == "10.12" && MacOS::Xcode.version >= "9.0"

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/libcurl.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/libcurl.pc
@@ -23,8 +23,8 @@
 # This should most probably benefit from getting a "Requires:" field added
 # dynamically by configure.
 #
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 supported_protocols="DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP"

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/libexslt.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/libexslt.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/libxml-2.0.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/libxml-2.0.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 modules=1

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/libxslt.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/libxslt.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/sqlite3.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/sqlite3.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/uuid.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/uuid.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 sharedlibdir=${libdir}
 includedir=${prefix}/include

--- a/Library/Homebrew/os/mac/pkgconfig/10.14/zlib.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.14/zlib.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 sharedlibdir=${libdir}
 includedir=${prefix}/include

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/libcurl.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/libcurl.pc
@@ -23,8 +23,8 @@
 # This should most probably benefit from getting a "Requires:" field added
 # dynamically by configure.
 #
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 supported_protocols="DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP"

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/libexslt.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/libexslt.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/libxml-2.0.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/libxml-2.0.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 modules=1

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/libxslt.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/libxslt.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/sqlite3.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/sqlite3.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/uuid.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/uuid.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 sharedlibdir=${libdir}
 includedir=${prefix}/include

--- a/Library/Homebrew/os/mac/pkgconfig/10.15/zlib.pc
+++ b/Library/Homebrew/os/mac/pkgconfig/10.15/zlib.pc
@@ -1,5 +1,5 @@
-prefix=/usr
-exec_prefix=${prefix}
+prefix=${homebrew_sdkroot}/usr
+exec_prefix=/usr
 libdir=${exec_prefix}/lib
 sharedlibdir=${libdir}
 includedir=${prefix}/include

--- a/Library/Homebrew/shims/mac/super/pkg-config
+++ b/Library/Homebrew/shims/mac/super/pkg-config
@@ -1,0 +1,5 @@
+#!/bin/sh
+
+exec "$HOMEBREW_OPT/pkg-config/bin/pkg-config" \
+  "--define-variable=homebrew_sdkroot=$HOMEBREW_SDKROOT" \
+  "$@"

